### PR TITLE
players on table real time + exit table

### PR DESCRIPTION
**Pull request by NaorNahman4**

### CODE REVIEW

File: server/controllers/tables.js [hunk 1]
```diff
@@ -13,8 +13,9 @@ const getAllTables = async (req, res) => {
 const validateTable = async (req, res) => {
     const name = req.body.name;
     const password = req.body.password
+    const username = req.body.username;
 
-    const table = await tableService.validateTable(name, password);
+    const table = await tableService.validateTable(name, password, username);
```
Comment (OrHaroni) [inline server/controllers/tables.js]: why need also uername ?

File: server/controllers/tables.js [hunk 2]
```diff
@@ -42,6 +43,20 @@ const addTable = async (req, res) => {
     }
 }
 
+// handeling the leave table request
+const leaveTable = async (req, res) => {
+    const username = req.body.username;
+    const name = req.body.name;
+    const status = await tableService.leaveTable(name,username);
```
Comment (OrHaroni) [inline server/controllers/tables.js]: not clear what is name...

File: server/models/tables.js [hunk 3]
```diff
@@ -34,17 +34,52 @@ const tableSchema = new Schema({
     type: [String],
     default: []
   },
-  playersOnTable: {
-    type: [String],
-    default: []
-  },
+   playersOnTable: [{
```
Comment (OrHaroni) [inline server/models/tables.js:37]: you had here a list of strings for all the nicknames of the users that in the table, if you want to add stuff that's good but no need username, password, email, isApprovedByEmail, LastTimeAddedMoney, NumberOfWins, gamesplayed...
Can do the same with nickname and moneyamount only... can maybe add "tableMoney" that is the total money he entered to the table with.

File: server/services/tables.js [hunk 4]
```diff
@@ -62,6 +69,23 @@ const addTable = async (table, userCreated) => {
     return 0; //everything good
 }
 
+const leaveTable = async (name,username) => {
+    try {
+        const table = await tableSchema.findOne({ "name": name });
+        const user = await userSchema.findOne({ "username": username });
```
Comment (OrHaroni) [inline server/services/tables.js:75]: again I dont understand what is name... also, nickname and username are distinct so you need also 1 of them to get 1 players (there are not 2 players with the same username or nickname) and remove them by it.

File: server/controllers/tables.js (hunk 1 above)
Comment (NaorNahman4) [reply server/controllers/tables.js]: to add him to the playersOnTable (Database) after the server check the valid of that table.

File: server/controllers/tables.js (hunk 2 above)
Comment (NaorNahman4) [reply server/controllers/tables.js]: name is the name of the table.

File: server/models/tables.js (hunk 3 above)
Comment (NaorNahman4) [reply server/models/tables.js:37]: you are right, maybe create a new schema of player or just the attributes needed

File: server/services/tables.js (hunk 4 above)
Comment (NaorNahman4) [reply server/services/tables.js:75]: name is the name of the table. we need it to find the correct table, username is the username of the user that leaving the table, we need to remove him from the playersOnTable . (i dont see nickname)

**Review by OrHaroni — APPROVED** [review]

**Pull request merged**